x-pack/plugins/enterprise_search/public/applications/analytics/components/analytics_collection_view/analytics_collection_events.test.tsx

```diff
@@ -7,31 +7,67 @@
 
 import '../../../__mocks__/shallow_useeffect.mock';
 
+import { setMockValues, setMockActions } from '../../../__mocks__/kea_logic';
+
 import React from 'react';
 
 import { shallow } from 'enzyme';
 
+import { EuiEmptyPrompt } from '@elastic/eui';
+
 import { AnalyticsCollection } from '../../../../../common/types/analytics';
 import { EntSearchLogStream } from '../../../shared/log_stream';
 
 import { AnalyticsCollectionEvents } from './analytics_collection_events';
 
 describe('AnalyticsCollectionEvents', () => {
-  const analyticsCollections: AnalyticsCollection = {
+  const analyticsCollection: AnalyticsCollection = {
     event_retention_day_length: 180,
     events_datastream: 'logs-elastic_analytics.events-example',
     id: '1',
     name: 'example',
   };
 
+  const mockActions = {
+    analyticsEventsIndexExists: jest.fn(),
+  };
+
   beforeEach(() => {
     jest.clearAllMocks();
+
+    setMockActions(mockActions);
   });
 
   it('renders', () => {
+    setMockValues({
+      isPresent: true,
+      isLoading: false,
+    });
     const expectedQuery = '_index: logs-elastic_analytics.events-example';
 
-    const wrapper = shallow(<AnalyticsCollectionEvents collection={analyticsCollections} />);
+    const wrapper = shallow(<AnalyticsCollectionEvents collection={analyticsCollection} />);
     expect(wrapper.find(EntSearchLogStream).prop('query')).toEqual(expectedQuery);
   });
+
+  describe('empty state', () => {
```

Comment:
Might be not an issue: using formatted message and testing it - could you make sure theres no console errors when running tests. I've encountered issues where FormattedMessage requires `mountWithI18nProvider` as requires i18n provider access in tests.